### PR TITLE
chore: refactor CM, CTB & i18n to accomodate option D&P

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds the option in the CTB to disable D&P
* refactors the EditView/Header to not show a status if it does not exist as a prop
* does not add the status column to the list-view if D&P is disabled
* does not display tabs or enable status query to manage tabs in the edit-view if D&P is disabled
* does not display the status in the locale picker from i18n

### Related issue(s)/PR(s)

* resolves CONTENT-2224
* resolves CONTENT-2232
* resolves CONTENT-2233
* resolves CONTENT-2234
* resolves CONTENT-2236
* resolves CONTENT-2237

### CODE REVIEW

File: packages/plugins/i18n/admin/src/components/CMHeaderActions.tsx [hunk 1]
```diff
@@ -5,12 +5,12 @@ import { useNotification, useQueryParams } from '@strapi/helper-plugin';
...
 import {
   type HeaderActionComponent,
-  unstable_useDocument,
+  unstable_useDocument as useDocument,
   unstable_useDocumentActions as useDocumentActions,
   type DocumentActionComponent,
 } from '@strapi/strapi/admin';
 import { useIntl } from 'react-intl';
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
```
Comment (jhoward1994) [inline packages/plugins/i18n/admin/src/components/CMHeaderActions.tsx:13]:
```suggestion
import { useNavigate } from 'react-router-dom';
```

**Review by jhoward1994 — COMMENTED** [review]
I think we discussed "Discard changes" also being a way to reset the CM form and not just to reset from modified status. If we do that I think non d&p content types will need access to that action too. Probably just something to keep in mind for now

**Comment by jhoward1994** [comment]
https://github.com/strapi/strapi/assets/48524071/db601e14-be45-4aae-b895-864c6f3ce47a

When you modify D&P from the CTB the app needs a reload before you can see the change in the FE

**Comment by joshuaellis** [comment]
> Screen.Recording.2024-03-11.at.10.05.55.mov 
> When you modify D&P from the CTB the app needs a reload before you can see the change in the FE

This is a bug with the CTB because the same behaviour is seen with enabling i18n, i've added it to the bug board for DX to work on :)

**Pull request merged**